cli/azd/pkg/azsdk/zip_deploy_client.go

```diff
@@ -113,8 +117,87 @@ func (c *ZipDeployClient) BeginDeploy(
 	return runtime.NewPoller(response, c.pipeline, pollerOptions)
 }
 
+// Deploys the specified application zip to the azure app service using deployment status api and waits for completion
+func (c *ZipDeployClient) BeginDeployTrackStatus(
+	ctx context.Context,
+	zipFile io.Reader,
+	subscriptionId,
+	resourceGroup,
+	appName string,
+	printStatus func(string),
+) (*runtime.Poller[armappservice.WebAppsClientGetProductionSiteDeploymentStatusResponse], error) {
+	printStatus("Creating deployment request")
+	request, err := c.createDeployRequest(ctx, zipFile)
+	if err != nil {
+		return nil, err
+	}
+
+	response, err := c.pipeline.Do(request)
+	if err != nil {
+		return nil, err
+	}
+
+	defer response.Body.Close()
+
+	if !runtime.HasStatusCode(response, http.StatusAccepted) {
+		return nil, runtime.NewResponseError(response)
+	}
+
+	client, err := armappservice.NewWebAppsClient(subscriptionId, c.cred, nil)
+	if err != nil {
+		return nil, fmt.Errorf("creating web app client: %w", err)
+	}
+
+	printStatus("Getting deployment status id")
+	deploymentStatusId := response.Header.Get("Scm-Deployment-Id")
+	matched, err := regexp.MatchString(`^[a-fA-F0-9]{8}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{12}$`, deploymentStatusId)
```

Comment:
Let's remove the guid validation -- right now the ID format is transparent to the client code (i.e., a format change doesn't break the client code).